src/hooks/__tests__/useNodeJsContributorsApi.test.tsx

```diff
@@ -1,148 +1,37 @@
-    localStorage.setItem('fetch_date', String(Date.now() + 5000));
-
-    fetchMock.mockResponse(JSON.stringify([createRandomContributorApiData()]));
-
-    render(<HookRenderer isVisible />);
-
-    await waitFor(() => {
-      expect(screen.getByText(/login_mock/i)).toBeInTheDocument();
-    });
-    await waitFor(() => {
-      expect(screen.getByText(/avatar_url_mock/i)).toBeInTheDocument();
-    });
-
-    // expect first call will be to random page instead of extra request for
-    // getting max pages
-    await waitFor(() => {
-      expect(
-        (fetchMock.mock.calls[0][0] as string).includes(
-          'https://api.github.com/repos/nodejs/node/contributors?per_page=1&page='
-        )
-      ).toBe(true);
-    });
-  });
-
-  it('should refetch expired data', async () => {
-    localStorage.setItem('fetch_date', '1');
-
-    fetchMock.mockResponses(
-      // mock first call to contributors API
-      [
-        '',
+import { fetchRandomContributor } from '../useNodeJsContributorsApi';
+
+describe('fetchRandomContributor', () => {
+  it('should return a random contributor', async () => {
+    window.fetch = jest.fn().mockResolvedValue({
+      headers: {
+        get: jest.fn().mockReturnValue('<link1>; rel="last"'),
+      },
+      json: jest.fn().mockResolvedValue([
         {
-          headers: {
-            link: linkResponseHeaderMock,
-          },
+          avatar_url: 'avatar_url',
+          login: 'login',
+          contributions: 123,
+          html_url: 'html_url',
         },
-      ],
-      // mock second request and return contributor mocked object
-      JSON.stringify([createRandomContributorApiData()])
-    );
-
-    render(<HookRenderer isVisible />);
-    await waitFor(() => {
-      expect(screen.getByText(/login_mock/i)).toBeInTheDocument();
-    });
-    await waitFor(() => {
-      expect(screen.getByText(/avatar_url_mock/i)).toBeInTheDocument();
-    });
-    await waitFor(() => {
-      expect(fetchMock.mock.calls[0][0]).toBe(
-        'https://api.github.com/repos/nodejs/node/contributors?per_page=1'
-      );
-    });
-    await waitFor(() => {
-      expect(
-        (fetchMock.mock.calls[1][0] as string).includes(
-          'https://api.github.com/repos/nodejs/node/contributors?per_page=1&page='
-        )
-      ).toBe(true);
-    });
-  });
-
-  describe('error handling', () => {
-    it('should return null on error', async () => {
-      fetchMock.mockReject(new Error('fake error message'));
-      render(<HookRenderer isVisible />);
-      const linkElement = screen.getByText(/null/i);
-      expect(linkElement).toBeInTheDocument();
-    });
-
-    it('should return null on unexpected link header from API', async () => {
-      fetchMock.mockResponseOnce('');
-      render(<HookRenderer isVisible />);
-      const linkElement = screen.getByText(/null/i);
-      expect(linkElement).toBeInTheDocument();
+      ]),
+    });
+    window.localStorage = {
```

Comment:
I believe we should test more scenarios here 🤔 otherwise this PR will decrease the code coverage of this hook by a bit

Reply:
I will try to cover as much as possible but I promise nothing